gcsmigrate/fids.py

```diff
@@ -4,14 +4,27 @@
 
 COLLECTION_NAME = os.environ["COLLECTION_NAME"]
 
+AT_ONCE = 10000
+
 
 def main():
     """Print all processed fids to stdout."""
     firestore_client = firestore.Client()
     collection = firestore_client.collection(COLLECTION_NAME)
-    docs = collection.select(["fid"]).stream()
-    for doc in docs:
-        print(doc.fid)
+    cursor = None
+    done = False
+    while not done:
+        if cursor is not None:
+            stream = (
+                collection.limit(AT_ONCE).order_by("fid").start_after(cursor).stream()
+            )
+        else:
+            stream = collection.limit(AT_ONCE).order_by("fid").stream()
```

Comment:
It's kind of annoying that we have to babysit this cursor.

Given the setting, this seems fine. But if we were to need to grab a large amount of records from firestore in the future, I wonder if exporting the collection into gcs, then loading from there (either into python, or into BigQuery) would be easier.

https://cloud.google.com/firestore/docs/manage-data/export-import
https://cloud.google.com/bigquery/docs/loading-data-cloud-firestore

Unfortunately, there doesn't seem to be a very convenient connector between firestore and Bigquery. Apparently you need to export the FS collection into GCS, then load that file into BQ.

Reply:
I should have documented this. I agree, it sucks. But without this, we get a timeout.
https://github.com/googleapis/nodejs-firestore/issues/964